docs/semantics.md

```diff
@@ -82,14 +87,26 @@ Warning: Using stat caching breaks the consistency guarantees discussed in this
 
 Because Cloud Storage does not forbid an object named ```foo``` from existing next to an object named ```foo/``` (see the Name conflicts section), when Cloud Storage FUSE is asked to look up the name "foo" it must stat both objects.
 
-Stat cache enabled with ```--stat-cache-ttl``` can help with this, but it does not help until after the first request. For example, assume that there is an object named foo but not one named ```foo/```, and the stat cache is enabled. When the user runs ```ls -l```, the following happens:
+Stat cache can help with this, but it does not help until after the first request. For example, assume that there is an object named foo but not one named ```foo/```, and the stat cache is enabled. When the user runs ```ls -l```, the following happens:
 - The objects in the bucket are listed. This causes a stat cache entry for ```foo``` to be created.
 - ```ls``` asks to stat the name ```foo```, causing a lookup request to be sent for that name.
 - Cloud Storage FUSE sends Cloud Storage stat requests for the object named ```foo``` and the object named ```foo/```. The first will hit in the stat cache, but the second will have to go all the way to Cloud Storage to receive a negative result.
 
 The negative result for ```foo/``` will be cached, but that only helps with the second invocation of ```ls -l```.
 
-To alleviate this, Cloud Storage FUSE supports a "type cache" on directory inodes. When ```--type-cache-ttl``` is set, each directory inode will maintain a mapping from the name of its children to whether those children are known to be files or directories or both. When a child is looked up, if the parent's cache says that the child is a file but not a directory, only one Cloud Storage object will need to be stated. Similarly if the child is a directory but not a file.
+To alleviate this, Cloud Storage FUSE supports a "type cache" on directory inodes. When type cache is enabled, each directory inode will maintain a mapping from the name of its children to whether those children are known to be files or directories or both. When a child is looked up, if the parent's cache says that the child is a file but not a directory, only one Cloud Storage object will need to be stated. Similarly if the child is a directory but not a file.
+
+The behavior of type cache is controlled by the following flags/config parameters:
+1. Type-cache capacity: The size of the type cache is **not configurable**. By default the type cache will hold up to 32768 items per directory. If you have folders containing more than 32768 items (folders or files) you may want to increase this, otherwise the caching will not function properly when listing that folder's contents:
```

Comment:
can we remove /1 for now and directly add whenever we make the type cache configurable ?

Reply:
Removed.